cmd/lvh/kernels/pull.go

```diff
@@ -0,0 +1,171 @@
+package kernels
+
+import (
+	"archive/tar"
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+	"regexp"
+	"runtime"
+	"strings"
+
+	"github.com/google/go-containerregistry/pkg/crane"
+	cranev1 "github.com/google/go-containerregistry/pkg/v1"
+	"github.com/spf13/cobra"
+)
+
+func pullCommand() *cobra.Command {
+	var platform string
+	var ci bool
+
+	cmd := &cobra.Command{
+		Use:   "pull <tag>",
+		Short: "pull built kernels by cilium/little-vm-helper-images",
+		Long: `Pull built kernels artifacts by cilium/little-vm-helper-images
+
+Examples:
+  # Pull latest main tag for version 6.6
+  lvh kernels pull 6.6-main
+
+  # Pull the bpf-next-20240404.012646 tag for version bpf-next
+  lvh kernels pull bpf-next-20240404.012646
+
+  # Pull the main tag for version 5.4 for arm64
+  lvh kernels pull 5.4-main --platform linux/arm64
+
+  # Pull the main tag for version 5.10 in directory mykernels
+  lvh kernels pull 5.10-main --dir mykernels
+
+  # Pull the latest tags available for version 5.15 without using 5.15-main
+  lvh kernels pull $(lvh kernels catalog 5.15 | tail -n 2 | head -n 1)`,
+		Args: cobra.ExactArgs(1),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			re := regexp.MustCompile(kernelTagRegex)
+			match := re.FindStringSubmatch(args[0])
+			if len(match) < 2 {
+				return fmt.Errorf("tag is malformed, it should follow %s: %s", re, args[0])
+			}
+
+			sPlatform := strings.SplitN(platform, "/", 2)
+			if len(sPlatform) < 2 {
+				return fmt.Errorf("platform is malformed, it must be <os>/<arch>: %s", platform)
+			}
+
+			repo := kernelImageRepository
+			if ci {
+				repo = fmt.Sprintf("%s-ci", repo)
+			}
+
+			srcImage := fmt.Sprintf("%s:%s", repo, args[0])
+			dstTarFile := fmt.Sprintf("%s.tar", args[0])
+
+			err := FetchTarImage(srcImage, dstTarFile, sPlatform[0], sPlatform[1])
+			defer os.Remove(dstTarFile) // this is on purpose before if != nil check
+			if err != nil {
+				return fmt.Errorf("failed fetching and creating tar image: %w", err)
+			}
+
+			pathToExtract := fmt.Sprintf("data/kernels/%s", match[1])
+
+			err = ExtractTarPath(dstTarFile, pathToExtract, filepath.Join(dirName, args[0]))
+			if err != nil {
+				return fmt.Errorf("failed extracting the file from the tar image: %w", err)
+			}
+			return nil
+		},
+	}
+
+	cmd.Flags().StringVarP(&platform, "platform", "p", runtime.GOOS+"/"+runtime.GOARCH, "platform for the kernel image <os>/<arch>")
+	cmd.Flags().StringVarP(&dirName, dirNameCommand, "d", ".", dirNameHelp)
```

Comment:
nit: Putting this into `.` by default seems a bit inconsistent with the other commands, I'd expect it to go under `_data/kernels/` by default. But I guess this is related to the "create directory" issue we discussed in the other thread.

Reply:
So the fact that we expect a certain hierarchy can create some weirdness as well, like this that I just bumped into right now, I needed to make the images folder an input in the LVH action because of a new runner requirements and then I have this:
<img width="1037" alt="image" src="https://github.com/cilium/little-vm-helper/assets/11256051/a82941ee-8123-40df-9e33-b4a27a8e39ed">